devices/DS18B20TemperatureSensor/main.c

```diff
@@ -0,0 +1,227 @@
+		at_ready = FALSE;
+	else {
+		sleep_time = (int)(response->value);
+		at_ready = TRUE;
+	}
+	
+	return XBEE_ATCMD_DONE;
+}
+
+/*
+ * Handles received API frames
+ * 
+ * xbee: a pointer to the sender XBee's information
+ * raw: a pointer to the packet struct containing the packet data
+ * length: the length of the information in raw, in bytes
+ * context: the context variable passed to the function; normally NULL
+ * 
+ * returns: 0 to indicate to the dispatcher it is finished processing the packet
+ */
+#ifdef ENABLE_XBEE_HANDLE_RX_SIMPLE_FRAMES
+int xbee_rx_simple_handler(xbee_dev_t *xbee, const void FAR *raw, uint16_t length, void FAR *context)
+{
+	const xbee_frame_receive_t FAR *frame = raw;
+	const int payload_length = length - sizeof(frame->frame_type) - sizeof(frame->ieee_address) - sizeof(frame->network_address_be) - sizeof(frame->options);
+	
+    if (strncmp(frame->payload, "Join confirmed", payload_length) == 0)
+    	rx_ready = TRUE;
+
+	return 0;
+}
+#endif
+
+/*
+ * Handles received modem status frames
+ * 
+ * xbee: a pointer to the sender (coordinator, in this case) XBee's information
+ * payload: the frame data
+ * length: the length of payload, in bytes
+ * context: the context variable passed to the function; normally NULL
+ */
+#ifdef ENABLE_XBEE_HANDLE_MODEM_STATUS_FRAMES
+int xbee_modem_status_handler(xbee_dev_t *xbee, const void FAR *payload,
+                              uint16_t length, void FAR *context)
+{
+	const xbee_frame_modem_status_t FAR *frame = payload;
+
+	if (frame->status == XBEE_MODEM_STATUS_JOINED)
+		joined = 1;
+	if (frame->status == XBEE_MODEM_STATUS_DISASSOC)
+		joined = 0;
+
+	return 0;
+}
+#endif
+
+/*
+ * Function to combine sys_watchdog_reset() and sys_xbee_tick() into one, since they are normally called together
+ */
+int tick_xbee()
+{
+	sys_watchdog_reset();
+	return sys_xbee_tick();
+}
+
+void main(void)
+{
+	ssize_t ret;
+	int16_t temperature;
+	wpan_envelope_t env;
+	int16_t handle;
+	sys_hw_init();
+	sys_xbee_init();
+	sys_app_banner();
+	
+	// Initialize the DS18B20 sensor
+	ret = ds18b20_config(NULL, DS18B20_TH_CFG, DS18B20_TL_CFG, DS18B20_RESOLUTION_CFG);
+	if (ret < 0)
+		printf("error");
+
+	// Loop while the XBee is not connected to a coordinator, since it cannot do much until that happens
+	printf("Waiting until joined...");
+	do {
+		tick_xbee();
+	} while (!joined);
+	printf("Joined!\n");
+
+	// Don't bother sending data until the coordinator is accepting data from this device
+	rx_ready = FALSE;
+	wpan_envelope_create(&env, &xdev.wpan_dev, WPAN_IEEE_ADDR_COORDINATOR, WPAN_NET_ADDR_UNDEFINED);
+	xbee_transparent_serial_str(&env, "Waiting to join...");
+	tick_xbee();
+	while (!rx_ready)
+	{
+		xbee_transparent_serial(&env);
+		delay_ticks(10);
+		tick_xbee();
+	}
+	printf("Coordinator accepted");
+	
+	// Wait 2 seconds to clear out any initialization commands that may still be queued
+	rtc_set_timeout(HZ * 2);
+	while (!rtc_timeout_expired())
+		tick_xbee();
+	
+	for (;;) {

```

Comment:
what does this do?  is it a while true?

Reply:
Yes, it's an infinite loop. It's the auto-generated infinite loop by CodeWarrior.